patch_rebaser/patch_rebaser.py

```diff
@@ -1,3 +1,164 @@
+    """Set up distro_info based on path"""
+    info_file, info_repo, remote = parse_distro_info_path(distroinfo_repo)
+
+    if remote:
+        di = info.DistroInfo(info_file, remote_git_info=info_repo)
+    else:
+        di = info.DistroInfo(info_file, local_info=info_repo)
+
+    return di.get_info()
+
+
+def get_patches_repo(distroinfo_repo, pkg_name, key):
+    """Get URL of repo with the patches branch"""
+    distro_info = get_distro_info(distroinfo_repo)
+    pkg = query.get_package(distro_info, pkg_name)
+    return pkg[key]
+
+
+def get_rebaser_config():
+    """Return a configparser object for patch_rebaser config"""
+    # Get the config file location based on path of currently running script
+    config_file = os.path.realpath(
+        "{0}/{1}".format(
+            os.path.dirname(os.path.realpath(__file__)),
+            "patch_rebaser.ini"
+        )
+    )
+    if not os.path.exists(config_file):
+        raise Exception(
+            "Configuration file {0} not found.".format(config_file)
+        )
+
+    rebaser_config = configparser.ConfigParser()
+    rebaser_config.read(config_file)
+    return rebaser_config
+
+
+def set_up_git_config(name, email):
+    os.environ["GIT_AUTHOR_NAME"] = name
+    os.environ["GIT_AUTHOR_EMAIL"] = email
+    os.environ["GIT_COMMITTER_NAME"] = name
+    os.environ["GIT_COMMITTER_EMAIL"] = email
+
+
+def main():
+    # These variables are set up by DLRN
+    user = os.environ['DLRN_USER']
+    local_repo = os.environ['DLRN_SOURCEDIR']
+    commit = os.environ['DLRN_SOURCE_COMMIT']
+    distroinfo_repo = os.environ['DLRN_DISTROINFO_REPO']
+    pkg_name = os.environ['DLRN_PACKAGE_NAME']
+
+    # The next variables come from patch_rebaser.ini
+    rebaser_config = get_rebaser_config()
+    remote = rebaser_config.get('DEFAULT', 'remote_name')
+    git_name = rebaser_config.get('DEFAULT', 'git_name')
+    git_email = rebaser_config.get('DEFAULT', 'git_email')
+    patches_repo_key = rebaser_config.get('distroinfo', 'patches_repo_key')
+
+    set_up_git_config(git_name, git_email)
+
+    repo = GitRepo(local_repo)
+
+    # Create a remote for the patches branch
+    patches_repo = get_patches_repo(
+        distroinfo_repo, pkg_name, patches_repo_key
+    )
+    if remote not in repo.remote.names():
+        if not repo.remote.add(remote, patches_repo):
+            raise Exception(
+                "Could not add remote {0} ({1})".format(remote, patches_repo)
+            )
+    repo.remote.fetch_all()
+
+    # Create local patches branch
+    branch = get_patches_branch(repo, remote, user)
+
+    # Not every project has a -patches branch for every release
+    if not branch:
+        # TODO: (future) Create and set up patches branch
+        return
+
+    remote_branch = "{remote}/{branch}".format(remote=remote, branch=branch)
+    repo.branch.create(branch, remote_branch, reset_if_exists=True)
+
+    # Rebase
+    LOGGER.info("Rebasing %s to %s", branch, commit)
```

Comment:
As we are targeting python 3.x, is there a reason to use this older style vs (ideally) f-strings, or at least .format()?

Reply:
Sweeet thank you for the pointer! Updated.

Reply:
For Python 2 I was always recommended to use %s and kwargs (no % in-between) for logging, to avoid formatting strings when unnecessary

Reply:
Hmm, I didn't see any references to this recommendation in the logging module for python 3.  I would advocate using the newer syntax and only reverting to something older if there is a demonstrable reason to do so.